modules/geo-layers/src/terrain-layer/terrain-layer.ts

```diff
@@ -343,6 +343,10 @@ export default class TerrainLayer<ExtraPropsT extends {} = {}> extends Composite
       );
     }
 
+    if (!this.state.terrain) {
+      return null;
+    }
```

Comment:
The `TerrainLayer#Empty Props` test was hanging. Maybe because it was creating a `SimpleMeshLayer` with an undefined mesh, and that is considered "not loaded"? `terrain` state is actually optional, but the type assumed it was always be defined.. I'm not sure why we'd create a layer when there isn't a anything to show

Reply:
`isTiled` will only be true when we have a non-Array form of `elevationData`. When `elevationData` is not defined, we have `isTiled = false` but `this.state.terrain` as `null` (due to `this.loadTerrain()`. In this case it does make sense to not render any sublayers, but I would say it is clearer to instead add the following to the start of `renderLayers()` instead:

```js
if (!elevationData) {
  return null;
}
```

If this check isn't triggered we can be sure that we will be can render a valid layer(s)

Reply:
Ok, I've pushed that up. I'm chasing some other failing tests related to tiles